final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,92 @@
+import arg_parser
+import feedparser
+import sys
+from urllib.request import Request, urlopen
+from urllib.error import URLError, HTTPError
+import dominate
+import html
+
+import news_parser
+import logger
+import get_cache
+import converter
+
+
+def main(channel, limit):
+    '''
+    1. cache news
+    2. create html or pdf document
+    3. or print in stdout news in json or normal format
+    '''
+    news_parser.cacheNews(args.source, channel)
+
+    if (args.tohtml or args.topdf):
+        html_document = dominate.document(title="HTML document")
+        converter.createHtmlStructure(channel, limit, html_document, args.tohtml, args.topdf)
+    else:
+        for index, item in enumerate(channel.entries):
+            if (index == limit):
+                break
+            logger.createLogs(item)
+
+            if (args.json):
+                print(news_parser.intoJson(item))
+            else:
+                print("\nTitle: ", html.unescape(item.title))
+                print("Date: ", item.published)
+                print("Link: ", item.link, '\n')
+                print("Description: ", news_parser.getDescription(item.description), '\n')
+                print("Links:", "\n[1]: ", item.link, "(link)\n[2]: ",
+                      news_parser.checkMediaContent(item), '\n')
+
+
+def checkArguments():
+    '''Check if in argument are --version or --help'''
+    if ('--version' in sys.argv or '--help' in sys.argv or '-h' in sys.argv):
+        return False
+    else:
+        return True
+
+
+def checkConnection(source):
+    '''Check connection to server'''
+    source = Request(source)
+    try:
+        response = urlopen(source)
+    except HTTPError as e:
+        print('The server couldn\'t fulfill the request.')
+        print('Error code: ', e.code)
+        logger.logging.error("HTTPError: " + str(e))
+        sys.exit()
```

Comment:
Here and below: It is better to have only one exit point in the application. It is better to raise custom exception here and then handle it in the main function.

Reply:
Fixed